rotkehlchen/chain/ethereum/modules/compound/v3/decoder.py

```diff
@@ -70,10 +101,151 @@ def decode_reward_claim(self, context: DecoderContext) -> DecodingOutput:
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.notes = f'Withdraw {amount} {underlying_token.symbol} from compound v3'
+                    action_from_event_type = HistoryEventType.SPEND
+                    action_to_event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    action_to_notes = f'Return {amount} {compound_token.symbol} to compound v3'
+
+                event.counterparty = CPT_COMPOUND_V3
+                paired_event = event
+                break
+
+        if (
+            paired_event is None or
+            action_from_event_type is None or action_to_event_subtype is None
+        ):
+            log.debug(
+                'At compound v3 supply/withdraw decoding of tx '
+                f'{context.transaction.tx_hash.hex()} the action item data was not found.',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        # also create an action item for the receive/return of the cTokens
+        action_item = ActionItem(
+            action='transform',
+            from_event_type=action_from_event_type,
+            from_event_subtype=HistoryEventSubType.NONE,
+            asset=compound_token,
+            to_event_subtype=action_to_event_subtype,
+            to_notes=action_to_notes,
+            to_counterparty=CPT_COMPOUND_V3,
+            paired_event_data=(paired_event, paired_event.event_type == HistoryEventType.DEPOSIT),
+        )
+        return DecodingOutput(action_items=[action_item])
+
+    def _correct_supply_event(
+            self,
+            decoded_events: list['EvmEvent'],
+            transaction: 'EvmTransaction',  # pylint: disable=unused-argument
+            all_logs: list['EvmTxReceiptLog'],  # pylint: disable=unused-argument
+    ) -> list['EvmEvent']:
+        """Post decoding function to correct supply event's receive amount in notes."""
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.RECEIVE_WRAPPED and
+                event.counterparty == CPT_COMPOUND_V3 and event.notes is not None
+            ):
+                event.notes = event.notes.format(amount=event.balance.amount)  # set the amount
+                break
+        return decoded_events
+
+    def _decode_repay_or_borrow_event(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a compound v3 repay or borrow event. Takes only the decoder context."""
+        for event in context.decoded_events:
+            if (
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset.identifier == evm_address_to_identifier(
+                    address=hex_or_bytes_to_address(context.tx_log.topics[3]),
+                    chain_id=self.evm_inquirer.chain_id, token_type=EvmTokenKind.ERC20,
```

Comment:
Each arg on its own line please if split

```suggestion
                    chain_id=self.evm_inquirer.chain_id,
                    token_type=EvmTokenKind.ERC20,
```